app/controllers/api/databases.php

```diff
@@ -3439,6 +3440,282 @@ function updateAttribute(
+    ->label('audits.resource', 'database/{request.databaseId}/collection/{request.collectionId}/document/{response.$id}')
+    ->label('usage.metric', 'documents.{scope}.requests.update')
+    ->label('usage.params', ['databaseId:{request.databaseId}', 'collectionId:{request.collectionId}'])
+    ->label('abuse-key', 'ip:{ip},method:{method},url:{url},userId:{userId}')
+    ->label('abuse-limit', APP_LIMIT_WRITE_RATE_DEFAULT * 2)
+    ->label('abuse-time', APP_LIMIT_WRITE_RATE_PERIOD_DEFAULT)
+    ->label('sdk.auth', [APP_AUTH_TYPE_SESSION, APP_AUTH_TYPE_KEY, APP_AUTH_TYPE_JWT])
+    ->label('sdk.namespace', 'databases')
+    ->label('sdk.method', 'incrementDocumentAttribute')
+    ->label('sdk.description', '/docs/references/databases/increment-document-attribute.md')
+    ->label('sdk.response.code', Response::STATUS_CODE_OK)
+    ->label('sdk.response.type', Response::CONTENT_TYPE_JSON)
+    ->label('sdk.response.model', Response::MODEL_DOCUMENT)
+    ->label('sdk.offline.model', '/databases/{databaseId}/collections/{collectionId}/documents')
+    ->label('sdk.offline.key', '{documentId}')
+    ->param('databaseId', '', new UID(), 'Database ID.')
+    ->param('collectionId', '', new UID(), 'Collection ID.')
+    ->param('documentId', '', new UID(), 'Document ID.')
+    ->param('key', '', new Key(), 'Attribute Key.')
+    ->param('value', null, new Numeric(), 'Value to increment attribute by')
+    ->inject('requestTimestamp')
+    ->inject('response')
+    ->inject('dbForProject')
+    ->inject('events')
+    ->inject('mode')
+    ->action(function (string $databaseId, string $collectionId, string $documentId, string $key, int|float $value, ?\DateTime $requestTimestamp, Response $response, Database $dbForProject, Event $events, string $mode) {
+        if ($value <= 0) {
+            throw new Exception(Exception::DOCUMENT_INVALID_STRUCTURE, 'Value must be greater than 0');
+        }
+
+        $database = Authorization::skip(fn() => $dbForProject->getDocument('databases', $databaseId));
+
+        if ($database->isEmpty() || (!$database->getAttribute('enabled') && $mode !== APP_MODE_ADMIN)) {
+            throw new Exception(Exception::DATABASE_NOT_FOUND);
+        }
+
+        $collection = Authorization::skip(fn() => $dbForProject->getDocument('database_' . $database->getInternalId(), $collectionId));
+
+        if ($collection->isEmpty() || !$collection->getAttribute('enabled')) {
+            if (!($mode === APP_MODE_ADMIN && Auth::isPrivilegedUser(Authorization::getRoles()))) {
+                throw new Exception(Exception::COLLECTION_NOT_FOUND);
+            }
+        }
+
+        // Validate attribute type
+        $attribute = $collection->find('key', $key, 'attributes');
+        if (!$attribute) {
+            throw new Exception(Exception::DOCUMENT_INVALID_STRUCTURE, "Attribute '$key' not found");
+        }
+
+        $validator = match ($attribute->getAttribute('type')) {
+            Database::VAR_INTEGER => new Integer(),
```

Comment:
If we use the find method above,  can we filter "type" for integers and floats?

Reply:
That find method works differently than the database find so no we can't. Anyways, I think it's good to provide a clear error message that the attribute was found, but the type was incorrect.

Reply:
No, I don't think that [find](https://github.com/utopia-php/database/blob/4035d3f7e3393385eabc7816055047659c6fb4d3/src/Database/Document.php#L265) can filter like that.

Reply:
you are right 👍